src/main/scala/com/fulcrumgenomics/bam/OverlappingBasesConsensusCaller.scala

```diff
@@ -0,0 +1,250 @@
+    */
+  def call(template: Template): CorrectionStats = (template.r1, template.r2) match {
+    case (Some(r1), Some(r2)) if r1.mapped && r2.mapped && r1.matesOverlap.contains(true) => call(r1, r2)
+    case _ => CorrectionStats(0, 0, 0, 0)
+  }
+
+  /** Consensus calls the overlapping bases if and only if both ends map with at least one base overlapping.
+    *
+    * @param r1 the first read in the pair
+    * @param r2 the second read in the pair
+    * @return summary statistics about how many bases were examined and modified
+    */
+  def call(r1: SamRecord, r2: SamRecord): CorrectionStats = {
+    require(r1.mapped && r2.mapped && r1.paired && r2.paired && r1.name == r2.name && r1.matesOverlap.contains(true))
+
+    // Clear and resize the builders
+    r1Builders.foreach { builder =>
+      builder.clear()
+      builder.sizeHint(r1.length)
+    }
+    r2Builders.foreach { builder =>
+      builder.clear()
+      builder.sizeHint(r2.length)
+    }
+
+    val r1Bases = r1.bases
+    val r1Quals = r1.quals
+    val r2Bases = r2.bases
+    val r2Quals = r2.quals
+
+    // Initialize the iters
+    val r1Iter        = MateOverlappingReadAndRefPosIterator(r1, r2).buffered
+    val r2Iter        = MateOverlappingReadAndRefPosIterator(r2, r1).buffered
+    var r1LastReadPos = r1Iter.head.read - 1
+    var r2LastReadPos = r2Iter.head.read - 1
+    var r1OverlappingBases = 0
+    var r2OverlappingBases = 0
+
+    // add all the bases prior to the overlapping bases
+    r1BasesBuilder.addAll(r1Bases.take(r1LastReadPos))
+    r1QualsBuilder.addAll(r1Quals.take(r1LastReadPos))
+    r2BasesBuilder.addAll(r2Bases.take(r2LastReadPos))
+    r2QualsBuilder.addAll(r2Quals.take(r2LastReadPos))
+
+    // Walk through the iterators by reference position
+    while (r1Iter.hasNext && r2Iter.hasNext) {
+      val r1Head = r1Iter.head
+      val r2Head = r2Iter.head
+      if (r1Head.ref < r2Head.ref) {
+        r1BasesBuilder.addAll(r1Bases.slice(from=r1LastReadPos + 1, until=r1Head.read + 1))
+        r1QualsBuilder.addAll(r1Quals.slice(from=r1LastReadPos + 1, until=r1Head.read + 1))
+        r1LastReadPos = r1Head.read
+        r1Iter.next()
+      }
+      else if (r2Head.ref < r1Head.ref) {
+        r2BasesBuilder.addAll(r2Bases.slice(from=r2LastReadPos + 1, until=r2Head.read + 1))
+        r2QualsBuilder.addAll(r2Quals.slice(from=r2LastReadPos + 1, until=r2Head.read + 1))
+        r2LastReadPos = r2Head.read
+        r2Iter.next()
+      }
+      else { // matched reference bases, so consensus call
+        // add read bases from insertions
+        r1BasesBuilder.addAll(r1Bases.slice(from = r1LastReadPos + 1, until = r1Head.read))
+        r1QualsBuilder.addAll(r1Quals.slice(from = r1LastReadPos + 1, until = r1Head.read))
+        r2BasesBuilder.addAll(r2Bases.slice(from = r2LastReadPos + 1, until = r2Head.read))
+        r2QualsBuilder.addAll(r2Quals.slice(from = r2LastReadPos + 1, until = r2Head.read))
```

Comment:
These are invoked every time as far as I can see, even when the last base was also mapped to the reference.  So many empty slices created.